core/src/main/java/com/netflix/conductor/core/execution/WorkflowSweeper.java

```diff
@@ -48,36 +52,31 @@ public WorkflowSweeper(WorkflowExecutor workflowExecutor, WorkflowRepairService
         ConductorProperties properties, QueueDAO queueDAO) {
         this.properties = properties;
         this.queueDAO = queueDAO;
-        this.executorThreadPoolSize = properties.getSweeperThreadCount();
-        if (this.executorThreadPoolSize > 0) {
-            this.executorService = Executors.newFixedThreadPool(executorThreadPoolSize);
-            init(workflowExecutor, workflowRepairService);
-            LOGGER.info("Workflow Sweeper Initialized");
-        } else {
-            LOGGER.warn("Workflow sweeper is DISABLED");
-        }
+        this.executorThreadPoolSize = properties.getSweeperThreadCount() > 0 ? properties.getSweeperThreadCount() : 1;
+        this.executorService = Executors.newFixedThreadPool(executorThreadPoolSize);
+        init(workflowExecutor, workflowRepairService);
+        LOGGER.info("Workflow Sweeper Initialized");
     }
 
     public void init(WorkflowExecutor workflowExecutor, WorkflowRepairService workflowRepairService) {
         ScheduledExecutorService deciderPool = Executors.newScheduledThreadPool(1);
         deciderPool.scheduleWithFixedDelay(() -> {
-            try {
-                boolean disable = properties.isSweepDisabled();
-                if (disable) {
-                    LOGGER.info("Workflow sweep is disabled.");
-                    return;
+            if (!running.get()) {
+                LOGGER.debug("Instance disabled, skip workflow sweep");
```

Comment:
The comment may not be appropriate. There could be other scenarios where Spring calls `stop()` on the `Lifecycle`.  A generic comment would be better here.

Reply:
changed